packages/amplify-frontend-android/lib/configuration-manager.js

```diff
@@ -1,43 +1,61 @@
 const inquirer = require('inquirer');
 const constants = require('./constants');
 
-function init(context) {
-  context.print.info('Describe your project:');
-  return promptForConfiguration(context);
+async function init(context) {
+  normalizeInputParams(context);
+  await confirmConfiguration(context);
+  return context;
 }
 
 function onInitSuccessful(context) {
-  return new Promise((resolve) => {
-    resolve(context);
-  });
+  return context;
 }
 
-function configure(context) {
-  return promptForConfiguration(context);
+async function configure(context) {
+  normalizeInputParams(context);
+  await confirmConfiguration(context);
+  return context;
 }
 
-function promptForConfiguration(context) {
+function normalizeInputParams(context) {
+  let inputParams;
+  if (context.exeInfo.inputParams && context.exeInfo.inputParams[constants.Label]) {
+    inputParams = context.exeInfo.inputParams[constants.Label];
+  }
+  if (inputParams && inputParams.config) {
+    if (!inputParams.config.ResDir) {
+      throw new Error('The command line parameter for android frontend configuration is incomplete.');
+    }
+  }
+  context.exeInfo.inputParams[constants.Label] = inputParams;
+}
+
+async function confirmConfiguration(context) {
   if (!context.exeInfo.projectConfig[constants.Label]) {
     context.exeInfo.projectConfig[constants.Label] = {};
   }
   if (!context.exeInfo.projectConfig[constants.Label].config) {
     context.exeInfo.projectConfig[constants.Label].config = {};
   }
-  const { config } = context.exeInfo.projectConfig[constants.Label];
-  const configurationSettings = [
-    {
-      type: 'input',
-      name: 'ResDir',
-      message: 'Where is your Res directory: ',
-      default: config.ResDir || 'app/src/main/res',
-    },
-  ];
-
-  return inquirer.prompt(configurationSettings)
-    .then((answers) => {
-      config.ResDir = answers.ResDir;
-      return context;
-    });
+
+  const inputParams = context.exeInfo.inputParams[constants.Label];
+  if (inputParams) {
+    Object.assign(context.exeInfo.projectConfig[constants.Label], inputParams);
+  } else if (!context.exeInfo.inputParams.yes) {
```

Comment:
The CLI should be intuitive enough to ask the question if the corresponding config is not present. The users shouldn't have to explicitly specify a yes flag.
There should be validations as to whether the configs passed from the CLI as well to prevent any erroneous behavior.

Reply:
Conversely it should infer the default values when it is running in headless mode